hikari/files.py

```diff
@@ -955,10 +955,10 @@ async def _wrap_iter(self) -> typing.AsyncGenerator[typing.Any, bytes]:
             for i in range(0, len(self.data), _MAGIC):
                 yield self.data[i : i + _MAGIC]  # noqa: E203
 
-        elif aio.is_async_iterator(self.data) or inspect.isasyncgen(self.data):
+        elif isinstance(self.data, typing.AsyncIterator) or isinstance(self.data, typing.AsyncGenerator):
```

Comment:
I'm not *quite* sure these `isinstance` checks are legal here, but I'll check soon:tm: if they actually are.